tools/generate_datatypes.py

```diff
@@ -465,8 +465,23 @@ def iter_types(v):
         l = list(filter(lambda t: t.name in selected_types, l))
     if args.no_builtin:
         l = list(filter(lambda t: type(t) != BuiltinType, l))
+    # remove opaque type
+    l = list(filter(lambda t: t.name not in opaque_type_mapping, l))
     return l
 
+def iter_opaque_types(v):
```

Comment:
Since `iter_opaque_types` is nearly the same as `iter_types` you can remove `iter_opaque_types` and instead pass a parameter to `iter_types`.

e.g. `def iter_types(v, includeOpaque):` and use it in the filter correspondingly

Reply:
Changes done!